### PR TITLE
Add operator logic

### PR DESCRIPTION
**Pull request by hieronx**

### CODE REVIEW

File: src/interfaces/IERC20.sol [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.21;
+pragma solidity >=0.5.0;
```
Comment (hieronx) [inline src/interfaces/IERC20.sol:2]: Requested by Superform, this helps integrations use the interfaces without being locked to solidity 0.8.21

**Review by AStox — DISMISSED** [review]
I would add a redemption integration test for good measure too, but otherwise it looks good! 👍

**Comment by hieronx** [comment]
> I would add a redemption integration test

@AStox added, which actually caught a small issue :)

**Review by AStox — DISMISSED** [review]
Haha I might've missed that issue, but glad my suggestion caught it!

**Review by AStox — APPROVED** [review]
Nice 👍

**Review by ilinzweilin — APPROVED** [review]

**Pull request merged**